eng/pipelines/templates/steps/sync-directory.yml

```diff
@@ -17,12 +17,23 @@ steps:
     - pwsh: |
         git clone --branch ${{ parameters.BaseBranchName }} https://github.com/azure/${{ repo }}
         $repoPath = "${{ repo }}/${{ parameters.DirectoryToSync }}"
-
-        if (!(Test-Path $repoPath)) { mkdir $repoPath }
-        Remove-Item -v -r $repoPath
-        Copy-Item -v -r $(Build.SourcesDirectory)/${{ parameters.DirectoryToSync }} $repoPath
-        Get-ChildItem -r $repoPath
-
+        if ($PatchFilesLocation)
+        {
+          pushd ${{ repo }}
+          $results = Get-ChildItem $PatchFilesLocation | % { git am $_.FullName }
+          if ($results.Count -ne (Get-ChildItem $PatchFilesLocation).Count)
+          {
+            Write-Output "Failed to apply patch files"
+            exit 1
```

Comment:
We should set the Has Changes variable in this step instead of the step that produces the patches as the changes really depends on what is already in each of the language repos.